.github/actions/delete-deployments/action.yml

```diff
@@ -0,0 +1,56 @@
+name: delete-deployments
+
+description: Deletes the deployments created by a workflow run in the scope of the PR.
+
+inputs:
+  owner:
+    description: The owner of the repository
+    required: true
+  repo:
+    description: The name of the repository
+    required: true
+  branch-name:
+    description: The name of the branch associated with pull request.
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: delete-deployments
+      uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+      with:
+        script: |
+          const perPage = 100;
+          let page = 1;
+          let allDeployments = [];
+          while(true){
+          const deployments = await github.rest.repos.listDeployments({
+            owner: "${{ inputs.owner }}",
+            repo: "${{ inputs.repo }}",
+            ref: "${{ inputs.branch-name }}",
+            per_page: perPage,
+            page: page
+          });
+
+          allDeployments = [...allDeployments, ...deployments.data];
+          if (!deployments.data.length) {
+          break;
+          }
+          page++;
+          }
+
+          await Promise.all(
+            allDeployments.map(async (deployment) => {
+              await github.rest.repos.createDeploymentStatus({
+                owner: "${{ inputs.owner }}",
+                repo: "${{ inputs.repo }}",
+                deployment_id: deployment.id,
+                state: 'inactive'
```

Comment:
Why is this needed please? Can't we delete a deployment if it's not inactive?

Reply:
[Here](https://docs.github.com/en/rest/deployments/deployments?apiVersion=2022-11-28#delete-a-deployment:~:text=If%20the%20repository%20has%20more%20than%20one%20deployment%2C%20you%20can%20only%20delete%20inactive%20deployments.), referring this had to set the status of deployments to inactive in order to delete.